src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -407,45 +407,60 @@ public void updateStudent(String courseId, String email, String newName,
             String lastName = StringHelper.splitName(newName)[1];
             
             if (isEmailChanged) {
-                CourseStudent newCourseStudent = new CourseStudent(newEmail, newName, newGoogleId, newComments,
-                                                                   courseId, newTeamName, newSectionName);
-                newCourseStudent.setLastName(lastName);
-                newCourseStudent.setCreatedAt(courseStudent.getCreatedAt());
-                if (keepUpdateTimestamp) {
-                    newCourseStudent.setLastUpdate(courseStudent.getUpdatedAt());
-                }
-                
+                CourseStudent newCourseStudent = new CourseStudent(newEmail, newName, newGoogleId,
+                        newComments,
+                        courseId, newTeamName, newSectionName);
                 try {
-                    createStudent(new StudentAttributes(newCourseStudent), hasDocument);
+                    updateStudentChangeEmail(newCourseStudent, lastName, courseStudent, hasDocument,
+                            keepUpdateTimestamp, courseId, email);
                 } catch (EntityAlreadyExistsException e) {
                     StudentAttributes existingStudent = (StudentAttributes) e.existingEntity;
                     String error = ERROR_UPDATE_EMAIL_ALREADY_USED
                             + existingStudent.getName() + "/" + existingStudent.getEmail();
                     throw new InvalidParametersException(error);
                 }
```

Comment:
If i do that i can't use this `throw new InvalidParametersException(error);` line in catch block.
 
 Please take a look build log of commit 5d8aa88 there is error which says `java.lang.RuntimeException: Expected exception not detected.[]` . 

This is what happens if i don't throw the exception. Its because in `StudentsDbTest.testUpdateStudentWithoutDocument(StudentsDbTest.java:223)` there is a case where a new student created and they are trying to update his email with existing email and it expects to throw an exception. So i don't think i can move it to the extracted method and throw the exception also.

Any suggestions?